### PR TITLE
Intermediate Representation Generator

### PR DESCRIPTION
**Pull request by morganzellers**
## Checklist:
- [x] Did you sign the [Contributor License Agreement](https://cla-assistant.io/wwt/SwiftCurrent)?
- [ ] Is the linter reporting 0 errors?
- [ ] Did you comply with our [styleguide](https://github.com/wwt/SwiftCurrent/blob/main/.github/STYLEGUIDE.md)?
- [ ] Is there [adequate test coverage](https://github.com/wwt/SwiftCurrent/blob/main/.github/CONTRIBUTING.md#test-etiquette) for your new code?
- [ ] Does the CI pipeline pass?

### CODE REVIEW

**Comment by morganzellers** [comment]
Separating some things out in prep to get this past a spike phase

**Comment by morganzellers** [comment]
Notes for tomorrow: 

Need to check inside enums better
Need to be able to find parent name
Formatting output

**Comment by morganzellers** [comment]
Was able to better check embedded types and link conforming types with their parents in a rough pass sort of way.

Currently getting this output:

```
extensions conforming to WorkflowDecodable:
- [FooName]
protocols conforming to WorkflowDecodable:
- [FRToo]
- [FooToo]
structs conforming to WorkflowDecodable:
- [Namespace.MyType]
Checking FRToo...
structs conforming to FRToo:
- [LoginView]
Checking FooToo...
structs conforming to FooToo:
- [FooView]
```

From these:

```
final class FooName: View {
    var _workflowPointer: AnyFlowRepresentable?
    var body: some View { Text("Foo") }
}

extension FooName: FlowRepresentable, WorkflowDecodable {
    static var flowRepresentableName: String {
        "FooName"
    }

    static func metadataFactory(launchStyle: LaunchStyle, flowPersistence: @escaping (AnyWorkflow.PassedArgs) -> FlowPersistence) -> FlowRepresentableMetadata {
        FlowRepresentableMetadata(Self.self, launchStyle: launchStyle, flowPersistence: flowPersistence)
    }
}

protocol FRToo: FlowRepresentable, WorkflowDecodable {
}

protocol FooToo: FlowRepresentable, WorkflowDecodable {
}

enum Namespace {
    struct MyType: FlowRepresentable, WorkflowDecodable { /* ... */ }
}

protocol ScrewWithMorgan: WorkflowDecodable

struct Lulz: FlowRepresentable, ScrewWithMorgan { /* ... */ }

struct FooView: View, FooToo {
    static var flowRepresentableName: String {
        "FooView"
    }

    static func metadataFactory(launchStyle: LaunchStyle, flowPersistence: @escaping (AnyWorkflow.PassedArgs) -> FlowPersistence) -> FlowRepresentableMetadata {
        FlowRepresentableMetadata(Self.self, launchStyle: launchStyle, flowPersistence: flowPersistence)
    }

    var _workflowPointer: AnyFlowRepresentable?

    var body: some View {
        Text("Woo")
    }
}

struct LoginView: View, FRToo {
....
```

File: ExampleApps/SwiftUIExample/DataDriven/main.swift [hunk 1]
```diff
@@ -93,18 +94,25 @@ func print(_ types: [Type.ObjectType: [ConformingType]], for conformance: String
...
+        let jsonString = String(data: encodedDictionary, encoding: .utf8)
+        print(jsonString!)
+
+        var filePath = ""
+        let directories: [String] = NSSearchPathForDirectoriesInDomains(FileManager.SearchPathDirectory.documentDirectory, FileManager.SearchPathDomainMask.allDomainsMask, true)
+        let directory = directories[0]
+        filePath = directory.appending("/" + filename)
+        print("Local path = \(filePath)")
+
+        try jsonString?.write(toFile: filePath, atomically: false, encoding: String.Encoding.utf8)
```
Comment (Tyler-Keith-Thompson) [inline ExampleApps/SwiftUIExample/DataDriven/main.swift:109]: Suggestion: don't write this to a particular place, send it to stdout. That'll allow CLI users to put it wherever they want.

Comment (morganzellers) [reply ExampleApps/SwiftUIExample/DataDriven/main.swift]: Yes! That's much more useful; thanks!

**Comment by morganzellers** [comment]
Work here was moved to the `swiftcurrent-ir-generator` branch and a new Pull Request (#186) has been opened. Closing this PR.

**Pull request closed without merging**